### ISSUE TITLE
Install questions

### ISSUE THREAD

**Issue by the-butterfry** [MFornander/LightSwarm#18]
Do you have a pinout diagram anywhere for the Wemos/ESP? Or where do you have that defined? I'm interested in getting this setup going on a few bikes for next year's burn. Thinking of adding microphone sync for providing an off-car light show near an art car that syncs up with the beat. 

Got a few other ideas too for bikes that would be a nice addition. Maybe start with 'bike swarm' mode, with a way to detect signal strength to order them? I donno, I just want to get this flashed on this years bikes to see what you've got and see if this is a viable path.

**Comment by MFornander** [comment]
Let me grab a picture of the boards we made.  The FastLED library allows four strands to be driven but we only used the first two output pins, whatever they are.

Note that the latest PlatformIO + + ArduinoESP + FastLED has some flickering on our strands that were not there last year.  Still tracking that issue down.

**Comment by the-butterfry** [comment]
cool, thanks. mostly this is just a test here for me anyhow to see if it works on bikes.
Are you still actively working on it this year? I'll take a look and see how I can contribute if this setup works for my purposes.

EDIT: oh wow, you've also got a rotary encoder going to switch effects. Nice! Did that ever get fully implemented?

**Comment by MFornander** [comment]
Looks like we used D7 and D6:
![img_20180911_223115](https://user-images.githubusercontent.com/7187414/45402425-d3919480-b612-11e8-86f3-80f496628c60.jpg)

**Comment by MFornander** [comment]
Yep! Had the rotary encoder working but didn't use it on the Playa since we got too busy having fun :)

- Rotate switches between animations.
- Click toggles between slow, normal, and fast time.
- Click and hold broadcasts the current animation and speed to the swarm.

**Comment by MFornander** [comment]
Gavin was brewing on beat detection back in 2017 too and opened up a ticket for it.  We thought about using a MSGEQ7 equalizer connected to the ESP and some code on the lower bass band:
https://www.ebay.com/itm/MSGEQ7-breakout-board-7-band-graphic-equalizer-for-audio-for-Arduino-RPi-PIC-/301210655107

**Comment by the-butterfry** [comment]
Awesome, I'll give it a go. I dug up a D1 Mini in my bins of bits too.

..{crazy idea}..
What I'd really like to build is a universal backend for a bike swarm moving art piece + small group fun. I'm imagining basically what you've built, but clearly in a different physical format/layout. I'd love to roll up to the jelly with my peeps and sync up with it, or in audio reactive mode, any mutant and provide a secondary light show. When we're riding around, multiple bikes could act as a single 'screen' synced across all the bikes, not just x bikes displaying the same thing. 

Basically the swarm tech, you guys are using with simple controls, with an easy way of uploading or activating new patterns (app/buttons/web interface). Bring a bike light platform down to earth so dozens of groups can implement it. When groups meet on the playa the software takes over and negotiates a new pattern. 

I might be talking about a 'product' here I guess. The knowledge barrier is pretty high and specialized, even if it is just plugging in the chips/flashing and building the lighting. This is dragon territory for most folks.. But I think we're getting to the point with the libraries where we can do this, and with some fit and finish on the implementation this tech can be used by folks who may not have such a high nerd level. The nerds just have to stay focused for long enough on the fit and finish to make it usable. :) 

Maybe it's a BM Honoraria grant thing, where instead of building an art piece you go to, it's an art piece that you implement and participate in... 

..{/crazy idea}..
